frame/pallet-xvm/src/lib.rs

```diff
@@ -29,11 +30,62 @@ pub mod evm;
 #[cfg(feature = "wasm")]
 pub mod wasm;
 
-/// XVM context consist of unique ID and optional execution arguments.
+// TODO: remove later after solution is properly benchmarked
+// Just a arbitrary weight constant to avoid having ZERO weight in some parts of execution
+pub const PLACEHOLDER_WEIGHT: u64 = 1_000_000;
+
+/// TODO: This isn't an exhaustive list, only a few are listed
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, scale_info::TypeInfo)]
+pub enum XvmError {
+    VmNotRecognized,
+    EncodingFailure,
+    ContextConversionFailed,
+    OutOfGas,
+    ExecutionError(Vec<u8>),
+}
+
+// TODO: Currently our precompile/chain-extension calls rely on direcy `Call` usage of XVM pallet.
+// This is perfectly fine when we're just calling a function in other VM and are interested whether the call was
+// successful or not.
+//
+// Problem arises IF we want to get back arbitrary read value from the other VM - `DispatchResultWithPostInfo` isn't enough for this.
+// We need to receive back a concrete value back from the other VM.
+
 #[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, scale_info::TypeInfo)]
+pub struct XvmCallOk {
+    /// Output of XVM call. E.g. if call was a query, this will contain query response.
+    output: Vec<u8>,
+    /// Total consumed weight. This is in context of Substrate (1 unit of weight ~ 1 ps of execution time)
+    consumed_weight: u64,
+}
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, scale_info::TypeInfo)]
+pub struct XvmCallError {
+    /// Result of XVM call
+    // TODO: use XvmError enum from pallet? Perhaps that's a better approach. Or at least provide mapping?
+    error: XvmError,
+    /// Total consumed weight. This is in context of Substrate (1 unit of weight ~ 1 ps of execution time)
+    consumed_weight: u64,
+}
+
+pub type XvmResult = Result<XvmCallOk, XvmCallError>;
```

Comment:
pub type should probably have a rustdoc

Reply:
Added, although it's self-explanatory in a way.